### ISSUE TITLE
Expanding period options for 1m interval via for loop

### ISSUE THREAD

**Issue by rodrigobercini** [dpguthrie/yahooquery#32]
**Is your feature request related to a problem? Please describe.**
I wouldn't call it a problem, but rather a potential. Currently, the maximum period users can fetch for 1m interval is 7 days. But one can fetch data up to 1 month ago. Find requests examples below:

✔️ aapl.history(interval='1m',period='7d')
❌ aapl.history(interval='1m',period='20d') Error message
❌aapl.history(interval='1m',period='1mo') Error message
✔️aapl.history(start='2020-07-18', end='2020-07-25', interval='1m')
✔️aapl.history(start='2020-07-25', end='2020-08-01', interval='1m')
✔️aapl.history(start='2020-08-01', end='2020-08-08', interval='1m')
✔️aapl.history(start='2020-08-08', end='2020-08-15', interval='1m')

**Describe the solution you'd like**
The first solution that came to my mind was to design a for loop specifically for 1m granularity. The loop would send four requests (one for each week) and concatenate the results in a single dataframe.

**Describe alternatives you've considered**
I've tried to assemble a loop, but it doesn't seem very clean. I reckon that there must be a more pythonic way of doing this task.

```
from yahooquery import Ticker
from datetime import datetime, timedelta
import pandas as pd

today = datetime.today()
first_start = (today - timedelta(days=7))
first_end = today

aapl = Ticker('aapl')
df = aapl.history(start=first_start, end=first_end, interval='1m')

mov_start = (today - timedelta(days=14))
mov_end = first_start

for i in range(3):
  df = pd.concat([aapl.history(start=mov_start, end=mov_end, interval='1m'), df])
  mov_start = (mov_start - timedelta(days=7))
  mov_end = (mov_end - timedelta(days=7))

```
**Additional context**
This is a potential functionality that would allow users to fetch more 1m interval data  in a single request. With that being said, I'm not a big fan of ad hoc solutions and I'm not sure how that would impact the code in general.

My point is: I'm not sure if this should be implemented or if it should just be indicated as an ad hoc solution in the documentation.

What do you think?

**Comment by dpguthrie** [comment]
Thanks, Rodrigo.  I think this is, at the very least, good info to have in the documentation.  I do think, though, that it would be relatively easy to incorporate into the existing history method.  Also, I can't think of a different way outside of a loop.  The slightly harder problem would be to make those four requests asynchronous in addition to making asynchronous requests across multiple symbols when asynchronous=True.

The solution I came up with doesn't make those four requests asynchronously but will make each date range request asynchronous across multiple symbols.  Just testing 5 symbols, I was able to retrieve 1 month of one-minute interval data (just under 40K rows) in just over two seconds.  Also, the API wouldn't change:

```
>>> def new_functionality():
...     start = time.time()
...     tickers = Ticker('fb aapl amzn nflx goog', asynchronous=True)
...     df = tickers.history(interval='1m', period='1mo')
...     end = time.time()
...     print(end - start)
...     print(len(df))
...
>>> new_functionality()
2.3366284370422363
38775
```

I'll keep testing but I can probably have it with the next release.

**Comment by rodrigobercini** [comment]
Thanks for the quick implementation. I've done some tests and it seems that something is wrong with the loop. While trying to fetch the last 30 days for AAPL, the end date is being set to 7 days ago.

today = Aug 30th

![image](https://user-images.githubusercontent.com/18664907/91668338-6ff42080-eae2-11ea-81ef-479aaec6fa5f.png)

If I understood correctly, this is being caused by the fact that the list of dates is starting with today - 7. Inserting today at index 0 might solve this issue.

```
dates = [_convert_to_timestamp(today - timedelta(7*x)) for x in range(5)]
dates.insert(_convert_to_timestamp(today), 0)
```

Cheers!

**Comment by dpguthrie** [comment]
It looks like the issue is with the sorting and not the data retrieval.

The first time through that dates loop will be today's date:

```python
today = datetime.today()

# first iteration through loop is today - 0
_convert_to_timestamp(today - timedelta(7 * 0))
```

So, the data is actually there; it's just a problem with the sorting (I didn't do inplace=True).  If you sort your dataframe, you'll see the last date is 8/28.

**Comment by rodrigobercini** [comment]
That makes a lot of sense. Perhaps adding `df.sort_values(by='date', inplace=True)` would also make sense?

**Comment by dpguthrie** [comment]
The sorting issue should be fixed now in 2.2.8.
